v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,234 @@
+func (ri *ARMResourceImporter) getStatus(ctx context.Context, armID string, armMeta genruntime.ARMMetaObject) (genruntime.ConvertibleStatus, error) {
+	// Create a request to get the current state of the resource
+	req, err := ri.createRequest(ctx, armID, armMeta.GetAPIVersion())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create request to import ARM resource %s", armID)
+	}
+
+	// Execute the request
+	resp, err := ri.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to execute request to import ARM resource %s", armID)
+	}
+
+	if !azruntime.HasStatusCode(resp, http.StatusOK) {
+		return nil, azruntime.NewResponseError(resp)
+	}
+
+	klog.V(3).Infof("Request succeeded")
+
+	armStatus, err := genruntime.NewEmptyARMStatus(armMeta, ri.Scheme())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create empty ARM status for importing ARM resource %s", armID)
+	}
+
+	// Populate our Status from the response
+	if err := azruntime.UnmarshalAsJSON(resp, armStatus); err != nil {
+		return nil, errors.Wrapf(err, "unable to deserialize ARM response for importing ARM resource %s", armID)
+	}
+
+	// Convert the ARM shape to the Kube shape
+	status, err := genruntime.NewEmptyVersionedStatus(armMeta, ri.Scheme())
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to construct status object for resource: %s", armID)
+	}
+
+	//TODO: what if it's not ok
+	if s, ok := status.(genruntime.FromARMConverter); ok {
+		var knownOwner genruntime.ArbitraryOwnerReference
+		err = s.PopulateFromARM(knownOwner, reflecthelpers.ValueOfPtr(armStatus)) // TODO: PopulateFromArm expects a value... ick
+		if err != nil {
+			return nil, errors.Wrapf(err, "converting ARM status to Kubernetes status for resource %s", armID)
+		}
+	}
+	return status, nil
+}
+
+// createRequest constructs the request to GET the ARM resource
+func (ri *ARMResourceImporter) createRequest(ctx context.Context, armID string, apiVersion string) (*policy.Request, error) {
+	//urlPath = strings.ReplaceAll(urlPath, "{resourceId}", ari.armID)
+	//req, err := runtime.NewRequest(ctx, http.MethodGet, runtime.JoinPaths(rmConfig.Endpoint, urlPath))
+	req, err := azruntime.NewRequest(ctx, http.MethodGet, azruntime.JoinPaths(ri.serviceConfiguration.Endpoint, armID))
+	if err != nil {
+		return nil, err
+	}
+
+	requestQueryPart := req.Raw().URL.Query()
+	requestQueryPart.Set("api-version", apiVersion)
+	req.Raw().URL.RawQuery = requestQueryPart.Encode()
+
+	req.Raw().Header.Set("Accept", "application/json")
+
+	klog.V(3).Infof("Created request to GET %s", req.Raw().URL.String())
+	return req, nil
+}
+
+func (ri *ARMResourceImporter) Client() *azruntime.Pipeline {
+	return ri.client
+}
+
+func (ri *ARMResourceImporter) Config() cloud.ServiceConfiguration {
+	return ri.serviceConfiguration
+}
+
+func (ri *ARMResourceImporter) createBlankObjectFromID(armID *arm.ResourceID) (runtime.Object, error) {
```

Comment:
Maybe we can skip this and use `unstructured.Unstructured` type here?

Reply:
How would that be an improvement? I need to pull back the ARM payload from Azure, then use methods we're code generating on the api types (implementing `genruntime.ImportableResource`) to convert that status into a spec. As far as I can tell, throwing away the type information and using unstructured would just make that harder.